addons/hr_holidays/models/hr_accrual_plan.py

```diff
@@ -0,0 +1,243 @@
+# # -*- coding: utf-8 -*-
+
+import datetime
+import logging
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from dateutil.relativedelta import relativedelta
+from datetime import datetime, time
+
+_logger = logging.getLogger(__name__)
+
+
+class AccrualPlan(models.Model):
+    _name = "hr.accrual.plan"
+    _description = "Accrual Plan"
+
+    name = fields.Char('Accrual Plan', required=True)
+    accrual_ids = fields.One2many(
+        'hr.accrual', 'plan_id',
+        string='Accrual')
+    company_id = fields.Many2one('res.company', string='Company',
+                                 required=True, readonly=True, default=lambda self: self.env.company)
+    employee_ids = fields.One2many('hr.employee', 'accrual_plan_id')
+    allocation_ids = fields.One2many('hr.leave.allocation', 'accrual_plan_id')
+    employees_count = fields.Integer("Employees", compute='_compute_employees_count')
+
+    def _compute_employees_count(self):
+        for plan in self:
+            plan.employees_count = len(plan.employee_ids)
+
+    def _get_accrual_line(self, employee):
+        #Get the appropriate accrual line accorded to the start date of an employee
+        line = False
+        date_start_work = datetime.combine(employee._get_date_start_work(), time(0, 0, 0))
+        today = datetime.combine(fields.Date.today(), time(0, 0, 0))
+        for accrual_line in self.accrual_ids:
+            if today > date_start_work + accrual_line._get_start_after_delta():
+                if line:
+                    if (date_start_work + line._get_start_after_delta()) < (
+                            date_start_work + accrual_line._get_start_after_delta()):
+                        line = accrual_line
+                else:
+                    line = accrual_line
+        return line
```

Comment:
i'm not sure what you're doing here but it looks like a break is missing once line is found ?